solvebio/test/test_object.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+# from solvebio.resource import Vault
+
+from .helper import SolveBioTestCase
+
+
+class ObjectTests(SolveBioTestCase):
+
+    def test_object_paths(self):
+        vaults = self.client.Vault.all()
+        for vault in vaults:
+            for file_ in list(vault.ls().solve_objects())[:5]:
+                o_path, _ = self.client.Object.validate_path(file_.full_path)
+                self.assertEqual(o_path, file_.full_path)
+
+    def test_object_path_cases(self):
+
+        user = self.client.User.retrieve()
+        domain = user.account.domain
+        user_vault = '{0}:user-{1}'.format(domain, user.id)
+        test_cases = [
+            ['{0}:myVault'.format(domain), '{0}:myVault:/'.format(domain)],
+            ['acme:myVault', 'acme:myVault:/'],
+            ['myVault', '{0}:/myVault'.format(user_vault)],
+            ['acme:myVault:/uploads_folder', 'acme:myVault:/uploads_folder'],
+            ['acme:myVault:/uploads_folder', 'acme:myVault:/uploads_folder'],
+            ['acme:myVault/uploads_folder', 'acme:myVault:/uploads_folder'],
+            ['myVault:/uploads_folder', '{0}:myVault:/uploads_folder'.format(domain)],  # noqa
+            ['/uploads_folder', '{0}:/uploads_folder'.format(user_vault)],
+            [':/uploads_folder', '{0}:/uploads_folder'.format(user_vault)],
+            ['myVault/uploads_folder', '{0}:/myVault/uploads_folder'.format(user_vault)],  # noqa
```

Comment:
raise error here too? We do that thing where if there is no slash it raises an error.